packages/flutter_tools/test/general.shard/compute_dev_dependencies_test.dart

```diff
@@ -30,393 +29,110 @@ const String _dartBin = 'bin/cache/dart-sdk/bin/dart';
-    return Pub.test(
-      fileSystem: fileSystem,
-      logger: logger,
-      processManager: processManager,
-      platform: FakePlatform(),
-      botDetector: const FakeBotDetector(false),
-      stdio: FakeStdio(),
-    );
+  void writePackages(List<Package> graph) {
+    final Map<String, Object?> packageConfigMap = <String, Object?>{'configVersion': 2};
+    for (final Package package in graph) {
+      fileSystem.file(fileSystem.path.join(package.name, 'pubspec.yaml'))
+        ..createSync(recursive: true)
+        ..writeAsStringSync('''
+name: ${package.name}
+dependencies:
+${package.dependencies.map((String d) => '  $d: {path: ../$d}').join('\n')}
+dev_dependencies:
+${package.devDependencies.map((String d) => '  $d: {path: ../$d}').join('\n')}
+''');
+      ((packageConfigMap['packages'] ??= <Object?>[]) as List<Object?>).add(<String, Object?>{
+        'name': package.name,
+        'rootUri': '../../${package.name}',
+        'packageUri': 'lib/',
+        'languageVersion': '3.7',
+      });
+    }
+    fileSystem.file(fileSystem.path.join(graph.first.name, '.dart_tool', 'package_config.json'))
+      ..createSync(recursive: true)
+      ..writeAsStringSync(jsonEncode(packageConfigMap));
   }
 
-  test('no dev dependencies at all', () async {
-    // Simulates the following:
-    //
-    // # /my_app/pubspec.yaml
-    // name: my_app
-    // dependencies:
-    //   package_a:
-    //
-    // # /package_a/pubspec.yaml
-    // name: package_a
-    // dependencies:
-    //   package_b:
-    final ProcessManager processes = _dartPubDepsReturns('''
-    {
-      "root": "my_app",
-      "packages": [
-        {
-          "name": "my_app",
-          "kind": "root",
-          "dependencies": [
-            "package_a",
-            "package_b"
-          ],
-          "directDependencies": [
-            "package_a"
-          ],
-          "devDependencies": []
-        },
-        {
-          "name": "package_a",
-          "kind": "direct",
-          "dependencies": [
-            "package_b"
-          ],
-          "directDependencies": [
-            "package_b"
-          ]
-        },
-        {
-          "name": "package_b",
-          "kind": "transitive",
-          "dependencies": [],
-          "directDependencies": []
-        }
-      ]
-    }''', project: project);
+  Future<void> validateComputeGraph(
+    List<Package> graph,
+    List<String> exclusiveDevDependencies,
+  ) async {
+    writePackages(graph);
+    final FlutterProject project = FlutterProject.fromDirectoryTest(fileSystem.directory('my_app'));
+
+    final PackageConfig packageConfig = await loadPackageConfig(project.packageConfig);
 
-    final Set<String> dependencies = await computeExclusiveDevDependencies(
-      pub(processes),
-      project: project,
-      logger: logger,
+    final Map<String, Dependency> dependencies = computeTransitiveDependencies(
+      project,
+      packageConfig,
+      fileSystem,
+      FakeLogger(),
```

Comment:
Why was the logger disabled? It's useful for understanding failures.

Instead, I'd:
- Keep BufferLogger
- Add assertions that the logger is free of errors (or warnings or both) in this function.